perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -54,6 +80,107 @@
                   'num_machines': 1}
 
 
+JOB_FILE_TEMPLATE = """
```

Comment:
We should decouple changing the default of FIO benchmarks from this pr. This PR is about introducing a way to dynamically generate a job file via a template and use it for various scenarios. 

If it's easy to use the new way to generate the same job file as the test was using before as default, great. Otherwise, we should just retain those job files and focus this PR on its original purpose.

We can address the default behavior of this benchmark in a separate discussion/PR.